components/cloud/azure/src/azblob.rs

```diff
@@ -725,7 +725,7 @@ impl BlobStorage for AzureStorage {
     async fn put(
         &self,
         name: &str,
-        mut reader: PutResource,
+        mut reader: PutResource<'_>,
```

Comment:
any reason the key iter is not implemented to azure?

Reply:
Yet we don't have an Azure account for testing this... I think we can implement this in the future?